websocket.go

```diff
@@ -559,6 +626,25 @@ func (c *wsConn) handleWsConn(ctx context.Context) {
 			if err := c.sendRequest(req.req); err != nil {
 				log.Errorf("sendReqest failed (Handle me): %s", err)
 			}
+		case <-c.pongs:
+			if c.timeout > 0 {
+				if err := c.conn.SetReadDeadline(time.Now().Add(c.timeout)); err != nil {
+					log.Error("setting read deadline", err)
+				}
+			}
+		case <-timeoutCh:
+			if c.pingInterval == 0 {
+				// pings not running, this is perfectly normal
+				continue
+			}
+
+			c.writeLk.Lock()
+			if err := c.conn.Close(); err != nil {
+				log.Warnw("timed-out websocket close error", "error", err)
+			}
+			c.writeLk.Unlock()
+			log.Errorw("Connection timeout", "remote", c.conn.RemoteAddr())
+			return
```

Comment:
Ideally we wouldn't return here, instead ether the fact that we're closing the connection, or expiring read deadline would cause reader on `nextMessage` above to return some error, closing the `c.incoming`, and more nicely closing everything here.

But that reader doesn't care, and I can't find a way to close it more forcefully